Runtime/Triangulator.cs

```diff
@@ -1194,15 +1035,15 @@ private bool ValidatePair(int i, int j)
         [BurstCompile]
         private struct ConstrainEdgesJob : IJob
         {
-            private NativeReference<Status> status;
+            public NativeReference<Status> status;
             [ReadOnly]
-            private NativeArray<float2> outputPositions;
-            private NativeArray<int> triangles;
+            public NativeArray<float2> positions;
+            public NativeArray<int> triangles;
             [ReadOnly]
-            private NativeArray<int> inputConstraintEdges;
-            private NativeList<Edge> internalConstraints;
-            private NativeArray<int> halfedges;
-            private readonly int maxIters;
+            public NativeArray<int> inputConstraintEdges;
+            public NativeArray<Edge> internalConstraints;
+            public NativeArray<int> halfedges;
```

Comment:
I've once more checked tests locally and get errors with most of the tests:

```csharp
Unhandled log message: '[Exception] InvalidOperationException: The Unity.Collections.NativeList`1[System.Int32] has been declared as [WriteOnly] in the job, but you are reading from it.'. Use UnityEngine.TestTools.LogAssert.Expect
```

Probably due to some internal Unity bug, this should be `NativeList`, this resolves the issue

```csharp
           public NativeList<int> halfedges;
```


---

I think this is my last comment, after that we are good to go with the PR.

Reply:
Which version of burst/collections are you using? The code seems very reasonable to me.

Reply:
Yeah, I think it's a bug in the collections/burst packages that have since been fixed.